sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/sources/ForeachWriterSuite.scala

```diff
@@ -258,6 +258,48 @@ class ForeachWriterSuite extends StreamTest with SharedSQLContext with BeforeAnd
       query.stop()
     }
   }
+
+  testQuietly("foreach with abort") {
```

Comment:
This test is a copy/paste of `foreach with error` with some modification. I think the common parts can be extracted. As I see the only difference is the writer and abort check at the end.